pallets/accounts/src/lib.rs

```diff
@@ -986,5 +1151,193 @@ pub mod pallet {
+					if num_weeks_since_lock_frag >= lock_period_in_weeks {
+						num_weeks_since_lock_frag = lock_period_in_weeks;
+					}
+
+					let tickets_amount_to_withdraw = tickets_amount_per_week_at_current_price *
+						(num_weeks_since_lock_frag - last_withdraw_week);
+
+					let nova_amount_to_withdraw = nova_amount_per_week_at_current_price *
+						(num_weeks_since_lock_frag - last_withdraw_week);
+
+					log::trace!("Tickets available per week: {}", tickets_convertible_per_week);
+					log::trace!("NOVA available per week: {}", nova_convertible_per_week);
+					log::trace!(
+						"Weeks passed since FRAG was locked: {}",
+						num_weeks_since_lock_frag
+					);
+					log::trace!(
+						"Tickets available per week at current price: {}",
+						tickets_amount_per_week_at_current_price
+					);
+					log::trace!(
+						"NOVA available per week at current price: {}",
+						nova_amount_per_week_at_current_price
+					);
+					log::trace!("Tickets to be withdrawn: {}", tickets_amount_to_withdraw);
+					log::trace!("NOVA to be withdrawn: {}", nova_amount_to_withdraw);
+
+					let tickets_amount_to_mint: <T as pallet_assets::Config>::Balance =
+						tickets_amount_to_withdraw.saturated_into();
+					let nova_amount_to_deposit: <T as pallet_balances::Config>::Balance =
+						nova_amount_to_withdraw.saturated_into();
+
+					ensure!(tickets_amount_to_withdraw > 0, Error::<T>::SystematicFailure);
+					ensure!(nova_amount_to_withdraw > 0, Error::<T>::SystematicFailure);
+					let nova_old_balance =
+						pallet_balances::Pallet::<T>::free_balance(&account.clone());
+					ensure!(
+						nova_old_balance + nova_amount_to_deposit >=
+							<pallet_balances::Pallet<T> as Currency<T::AccountId>>::minimum_balance(
+							),
+						Error::<T>::SystematicFailure // this should never happen
+					);
+
+					ensure!(
+						!nova_old_balance.checked_add(&nova_amount_to_deposit).is_none(),
+						Error::<T>::SystematicFailure
+					);
+
+					// Checks completed, now write
+					// mint tickets
+					<pallet_assets::Pallet<T> as Mutate<T::AccountId>>::mint_into(
+						T::TicketsAssetId::get(),
+						&account.clone(),
+						tickets_amount_to_mint,
+					)?;
+
+					// Assign NOVA
+					// This will not fail if all the checks above passed, so the result can be ignored
+					let _ =
+						<pallet_balances::Pallet<T> as fungible::Mutate<T::AccountId>>::mint_into(
+							&account.clone(),
+							nova_amount_to_deposit,
+						)?;
+
+					// remove the locked frag information for this account
+					if num_weeks_since_lock_frag == lock_period_in_weeks {
+						<EthLockedFrag<T>>::remove(&eth_account);
+					} else {
+						// Update the week number of the latest withdraw
+						<EthLockedFrag<T>>::mutate(&eth_account, |eth_lock| {
+							let eth_lock = eth_lock.as_mut().unwrap();
+							eth_lock.block_number = <frame_system::Pallet<T>>::block_number();
```

Comment:
but this value should be immutable, is it the case when multiple locks happen actually the issue?

Reply:
Multiple lock is not considered at the moment.

Reply:
Implemented.